File: Sample Projects/VWMasterApplication/VWMasterApplication/VWMoPubServerViewController.m
//
//  VWMoPubServerViewController.m
//  VWMasterApplication
//
//  Created by Christina Long on 10/2/15.
//  Copyright © 2015 Christina Long. All rights reserved.
//

#import "VWMoPubServerViewController.h"
#import "MoPub.h"
#import "MPAdView.h"
#import "MPInterstitialAdController.h"

static NSString *const kBannerAdUnit = @"81e4362779af4f31b392223cece421a3";
static NSString *const kInlineAdUnit = @"12a8caf3a56e40a5a883b36f910f8d96";
static NSString *const kInterstitialAdUnit = @"400efb79cbfd4063a777d471ac25c795";
static NSString *const kLeaderboardAdUnit = @"83023bd000f44a12a02de151b1664e30";

@interface VWMoPubServerViewController () <MPAdViewDelegate, MPInterstitialAdControllerDelegate>

@property (strong, nonatomic) MPAdView *bannerAdView;
@property (strong, nonatomic) MPAdView *inlineAdView;
@property (strong, nonatomic) MPInterstitialAdController *interstitialAdView;

@end

@implementation VWMoPubServerViewController


#pragma mark - View Lifecycle

- (void)viewDidLoad {
    [super viewDidLoad];
    
    NSLog(@"MoPub SDK version: %@", [[MoPub sharedInstance] version]);
    
    [self addBannerAdView];
    [self addInlineAdView];
}


#pragma mark - Banner Ad

-(void)addBannerAdView {
    //determine size for device
    CGSize size = (UI_USER_INTERFACE_IDIOM() == UIUserInterfaceIdiomPad) ? MOPUB_LEADERBOARD_SIZE : MOPUB_BANNER_SIZE;
    NSString *adUnit = (UI_USER_INTERFACE_IDIOM() == UIUserInterfaceIdiomPad) ? kLeaderboardAdUnit : kBannerAdUnit;
    
    self.bannerAdView = [[MPAdView alloc] initWithAdUnitId:adUnit size:size];
    self.bannerAdView.delegate = self;
    self.bannerAdView.backgroundColor = [UIColor grayColor];
    
    //determine size/frame for banner Ad
    CGRect bounds = self.view.bounds;
    CGRect adFrame = CGRectZero;
    
    adFrame.size = [self.bannerAdView sizeThatFits:bounds.size];
    adFrame.origin.x = (bounds.size.width - adFrame.size.width)/2;
    adFrame.origin.y = bounds.size.height - adFrame.size.height - self.tabBarController.tabBar.frame.size.height;
    
    self.bannerAdView.frame = adFrame;
    
    //ad the banner ad to the view
    [self.view addSubview:self.bannerAdView];
}

- (IBAction)requestBannerAd:(id)sender {
    [self.bannerAdView loadAd];
}


#pragma mark - Inline Ad

-(void)addInlineAdView {
    //add the inline ad view
    self.inlineAdView = [[MPAdView alloc] initWithAdUnitId:kInlineAdUnit size:MOPUB_MEDIUM_RECT_SIZE];
    self.inlineAdView.delegate = self;
    self.inlineAdView.backgroundColor = [UIColor grayColor];
    
    //determine frame and size for the view
    CGRect bounds = self.view.bounds;
    CGRect adFrame = CGRectZero;
    
    adFrame.size = [self.inlineAdView sizeThatFits:bounds.size];
    self.inlineAdView.frame = CGRectMake((bounds.size.width - adFrame.size.width)/2, (bounds.size.height - adFrame.size.height)/2, adFrame.size.width, adFrame.size.height);
    
    //add the inline ad view
    [self.view addSubview:self.inlineAdView];
}

- (IBAction)requestInlineAd:(id)sender {
    [self.inlineAdView loadAd];
}


#pragma mark - Interstitial Ad

- (IBAction)requestInterstitialAd:(id)sender {
    if (!self.interstitialAdView) {
        self.interstitialAdView = [MPInterstitialAdController interstitialAdControllerForAdUnitId:kInterstitialAdUnit];
        self.interstitialAdView.delegate = self;
    }
    
    [self.interstitialAdView loadAd];
}


#pragma mark - MPAdViewDelegate

- (void)adViewDidLoadAd:(MPAdView *)view {
    NSLog(@"adViewDidReceiveAd:");
}

- (void)adViewDidFailToLoadAd:(MPAdView *)view {
    NSLog(@"adView:didFailToReceiveAdWithError:");
}

- (UIViewController *)viewControllerForPresentingModalView {
    NSLog(@"viewControllerForPresentingModalView");
    return self;
}

- (void)willPresentModalViewForAd:(MPAdView *)view {
    NSLog(@"willPresentModalViewForAd:");
}

- (void)didDismissModalViewForAd:(MPAdView *)view {
    NSLog(@"didDismissModalViewForAd:");
}

- (void)willLeaveApplicationFromAd:(MPAdView *)view {
    NSLog(@"willLeaveApplicationFromAd:");
}


#pragma mark - MPInterstitialAdControllerDelegate

- (void)interstitialDidLoadAd:(MPInterstitialAdController *)interstitial {
    NSLog(@"interstitialDidLoadAd:");
    
    [interstitial showFromViewController:self];
}

- (void)interstitialDidFailToLoadAd:(MPInterstitialAdController *)interstitial {
    NSLog(@"interstitialDidFailToLoadAd:");
}

- (void)interstitialWillAppear:(MPInterstitialAdController *)interstitial {
    NSLog(@"interstitialWillAppear:");
}

- (void)interstitialDidAppear:(MPInterstitialAdController *)interstitial {
    NSLog(@"interstitialDidAppear:");
}

- (void)interstitialWillDisappear:(MPInterstitialAdController *)interstitial {
    NSLog(@"interstitialWillDisappear:");
}

- (void)interstitialDidDisappear:(MPInterstitialAdController *)interstitial {
    NSLog(@"interstitialDidDisappear:");
}

- (void)interstitialDidExpire:(MPInterstitialAdController *)interstitial {
    NSLog(@"interstitialDidExpire:");
}

- (void)interstitialDidReceiveTapEvent:(MPInterstitialAdController *)interstitial {
    NSLog(@"interstitialDidReceiveTapEvent:");
}

@end

File: Sample Projects/VWAdSuite/VWAdSwiftExample/VWAdSwiftExample/VWMoPubServerViewController.swift
//
//  VWMoPubServerViewController.swift
//  VWAdSwiftExample
//
//  Created by Christina Long on 10/5/15.
//  Copyright © 2015 Verve Wireless. All rights reserved.
//

import UIKit

class VWMoPubServerViewController: UIViewController, MPAdViewDelegate, MPInterstitialAdControllerDelegate {
  let kBannerAdUnit : String = "81e4362779af4f31b392223cece421a3"
  let kInlineAdUnit : String = "12a8caf3a56e40a5a883b36f910f8d96"
  let kInterstitialAdUnit : String = "400efb79cbfd4063a777d471ac25c795"
  
  var bannerAdView : MPAdView?
  var inlineAdView : MPAdView?
  var interstitialAd : MPInterstitialAdController?
  
  
  //MARK: View Lifecycle
  
  override func viewDidLoad() {
    super.viewDidLoad()
    
    addBannerAdView()
    addInlineAdView()
  }
  
  
  //MARK: Banner Ad
  
  func addBannerAdView() {
    let adSize: CGSize = UI_USER_INTERFACE_IDIOM() == .Pad ? MOPUB_LEADERBOARD_SIZE : MOPUB_BANNER_SIZE
    
    bannerAdView = MPAdView.init(adUnitId: kBannerAdUnit, size: adSize)
    bannerAdView?.delegate = self
    bannerAdView?.backgroundColor = UIColor.grayColor()
    
    let bounds : CGRect = view.bounds
    var adFrame : CGRect = CGRectZero
    
    adFrame.size = bannerAdView!.sizeThatFits(bounds.size)
    adFrame.origin.x = (bounds.size.width-adFrame.size.width)/2
    adFrame.origin.y = bounds.size.height - adFrame.size.height - tabBarController!.tabBar.frame.size.height
    
    bannerAdView!.frame = adFrame
    
    self.view?.addSubview(bannerAdView!)
  }
  
  @IBAction func requestBannerAd() {
    bannerAdView?.loadAd()
  }
  
  
  //MARK: Inline Ad
  
  func addInlineAdView() {
    if inlineAdView == nil {
      inlineAdView = MPAdView.init(adUnitId: kInlineAdUnit, size: MOPUB_MEDIUM_RECT_SIZE)
      inlineAdView?.delegate = self
      inlineAdView?.backgroundColor = UIColor.grayColor()
      
      let bounds : CGRect = view.bounds
      var adFrame : CGRect = CGRectZero
      
      adFrame.size = inlineAdView!.sizeThatFits(bounds.size)
      inlineAdView!.frame = CGRectMake((bounds.size.width - adFrame.size.width)/2, (bounds.size.height - adFrame.size.height)/2, adFrame.size.width, adFrame.size.height);
      
      self.view?.addSubview(inlineAdView!)
    }
  }
  
  @IBAction func requestInlineAd() {
    inlineAdView?.loadAd()
  }
  
  
  //MARK: Interstitial Ad
  
  @IBAction func requestInterstitialAd() {
    interstitialAd = MPInterstitialAdController.init(forAdUnitId: kInterstitialAdUnit)
    interstitialAd?.delegate = self
    interstitialAd?.loadAd()
  }
  
  
  //MARK: MPAdViewDelegate
  
  func adViewDidLoadAd(view: MPAdView!) {
    if (view != nil) {
      NSLog("adViewDidLoadAd: %@", view)
    }
  }
  
  func  adViewDidFailToLoadAd(view: MPAdView!) {
    if (view != nil) {
      NSLog("adViewDidFailToLoadAd: %@", view)
    }
  }
  
  func viewControllerForPresentingModalView() -> UIViewController! {
    return self
  }
  
  func willPresentModalViewForAd(view: MPAdView!) {
    if (view != nil) {
      NSLog("willPresentModalViewForAd: %@", view)
    }
  }
  
  func didDismissModalViewForAd(view: MPAdView!) {
    if (view != nil) {
      NSLog("didDismissModalViewForAd: %@", view)
    }
  }
  
  func willLeaveApplicationFromAd(view: MPAdView!) {
    if (view != nil) {
      NSLog("willLeaveApplicationFromAd: %@", view)
    }
  }
  
  
  //MARK: MPInterstitialAdControllerDelegate
  
  func interstitialDidLoadAd(interstitial: MPInterstitialAdController!) {
    if (interstitial != nil) {
      NSLog("interstitialDidLoadAd: %@", interstitial)
      
      interstitial.showFromViewController(self)
    }
  }
  
  func interstitialDidFailToLoadAd(interstitial: MPInterstitialAdController!) {
    if (interstitial != nil) {
      NSLog("interstitialDidFailToLoadAd: %@", interstitial)
    }
  }
}
